### DISCUSSION TITLE
在vue-shape中自定义节点组件不生效

### DISCUSSION THREAD

**Discussion by xiong-universe** [antvis/X6#2727] [Q&A]
自定义vue节点，然后注册组件时，子组件引用element组件不生效，浏览器报错 [Vue warn]: Failed to resolve component: el-button，正常使用element时可以生效，请问这是书写方式的问题吗？
Graph.registerVueComponent(
        "img-node",
        {
          template: `<node/>`,
          components: {
            Node,
          },
        },
        true
      );
      graph.addNode({
        id: "1",
        shape: "vue-shape",
        component: "img-node",
        x: 400,
        y: 150,
        width: 150,
        height: 100,
        data: {
          num: 0,
        },
      });

**Comment by lloydzhou** [comment]
默认情况vue-shape会生成一个新的vue App挂载到节点内部。这个新的App不能识别`el-button`

1. 你可以在你自己定义的Node组件内部使用Button
```
import { Button } from 'element-ui';
然后在模板里面使用
<Button>确定</Button>
```
2. 如果你使用vue3的话，你还可以使用useTeleport就不会生成那么多的新的Vue App，这样就可以用到全局vue的一些配置。
